3rdParty/rocksdb/v5.1.4/util/env_encryption.cc

```diff
@@ -577,8 +802,8 @@ Status CTRCipherStream::EncryptBlock(uint64_t blockIndex, char *data, char* scra
 
   // Create nonce + counter
   auto blockSize = cipher_.BlockSize();
-  memmove(scratch, iv_, blockSize);
-  *((uint64_t*)scratch) = blockIndex;
+  memmove(scratch, iv_.data(), blockSize);
+  EncodeFixed64(scratch, blockIndex + initialCounter_);
```

Comment:
So the goal is to produce a 16-byte nonce. It's not entirely clear to me what is happening here. To my understanding, we should be taking an 8-byte IV and an 8-byte counter value and appending the two. Is that what is happening here? Is `iv_` 16 bytes and we're overwriting the first 8 bytes of it with the counter?